fastestimator/dataset/brats.py

```diff
@@ -0,0 +1,274 @@
+        mean_sample_wise = []
+        std_sample_wise = []
+        for mean_item, std_item in results:
+            mean_sample_wise.append(mean_item)
+            std_sample_wise.append(std_item)
+        mean_sample_wise = np.asarray(mean_sample_wise)
+        std_sample_wise = np.asarray(std_sample_wise)
+        std_samples = np.mean(std_sample_wise, axis=0, dtype=np.float32)
+        mean_samples = np.mean(mean_sample_wise, axis=0, dtype=np.float32)
+
+        path_brats_preprocessed = [path_brats_preprocessed] * len(samples)
+        mean_samples = [mean_samples] * len(samples)
+        std_samples = [std_samples] * len(samples)
+        resized_img_shape_list = [resized_img_shape] * len(samples)
+        num_cpu = mp.cpu_count()
+        pool = Pool(processes=num_cpu)
+        pool.starmap(_generate_preprocessed_samples_core,
+                     zip(path_brats_preprocessed, samples, mean_samples, std_samples, resized_img_shape_list))
+
+    train_val_indices = np.arange(len(samples))
+    shuffle(train_val_indices)
+    n_training = int(len(samples) * val_split)
+    train_indices = train_val_indices[:n_training]
+    val_indices = train_val_indices[n_training:]
+
+    hgg_samples = glob(os.path.join(path_brats, 'data', 'HGG', '*'))
+    lgg_samples = glob(os.path.join(path_brats, 'data', 'LGG', '*'))
+    samples = lgg_samples + hgg_samples
+
+    samples_train = [samples[idx] for idx in train_indices]
+    train_csv = _create_csv(path_brats, 'train', samples_train)
+
+    samples_val = [samples[idx] for idx in val_indices]
+    val_csv = _create_csv(path_brats, 'val', samples_val)
+    return train_csv, val_csv
+
+
+def _crop_to_non_zero_content(modality_filenames):
+    tolerance = 1e-5
+    background = 0
+    # determines non zero region across t1,t2,t1ce,flair images, crops the non-zero region
+    data_valid = None
+    for idx, mod_fn in enumerate(modality_filenames):
+        data_nifti = nib.load(mod_fn)
+        data = data_nifti.get_data()
+        if idx == 0:
+            data_valid = np.zeros(data.shape)
+        data_valid_idx = np.logical_or(data < (background - tolerance), data > (background + tolerance))
+        data_valid[data_valid_idx] = 1
+    coords = np.array(np.where(data_valid))
+    start = coords.min(axis=1)
+    end = coords.max(axis=1) + 1
+    start = np.maximum(start - 1, 0)
+    end = np.minimum(end + 1, data_valid.shape)
+    slices = [slice(s, e) for s, e in zip(start, end)]
+    return slices
+
+
+def load_data(path_brats=None, resized_img_shape=(144, 144, 144), bias_correction=False):
+    assert path_brats is not None, "path_brats should have valid directory 'data' where braTs 2018 dataset has been downloaded  having 'LGG' and 'HGG' subdirs"
+    home = str(Path.home())
+    fe_data = os.path.join(home, 'fastestimator_data')
+    assert fe_data == os.path.dirname(path_brats), "Brats data is expected to be sub directory in ~/fastestimator_data"
```

Comment:
we shouldn't require this, it should just be our default expectation (where we look if path is None). I think all our other datasets can technically be forced into other folders so long as the directory location is given.

Reply:
Now it  checks for the path is None , if None  then go for default locaiton ~/fastestimator_data/BraTs